### PR TITLE
CONTRIBUTING.md update for docs

### PR DESCRIPTION
**Pull request by AhmetCanSolak**
# Description

It is a small PR to update contributing guidelines about how to benefit from and update the dev-docs.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# References

follow-up of #99 .

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [X] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by kne42 — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -45,6 +59,13 @@ You can view them with:
 $ git status
 ```
 
+If you are changing anything with files and folder structure, 
+remember to update doctree
+```sh
+$ cd docs
+$ sphinx-apidoc -f -P -F -o source ../napari
```
Comment (jni) [inline CONTRIBUTING.md:66]: Should this instruction not just be part of the build process? My guide for what needs to be committed in git is that *anything generated* should not be committed. I think @royerloic was running into this problem yesterday after rebasing his dimensions branch.

**Review by jni — COMMENTED** [review]
@AhmetCanSolak just a question, I won't let it be blocking. =)

File: CONTRIBUTING.md (hunk 1 above)
Comment (AhmetCanSolak) [reply CONTRIBUTING.md:66]: @jni It depends. This is only one way of generating doc source from the docstrings. IMO, sphinx commands can easily result in different ways, so enforcing single way of generating dev-docs sounds good to me for consistency. We can try and see if this brings any problems, also love to see what problem @royerloic had. 

Also, if we do not want to have people update doc source with their PRs, we need to have doc source update PRs regularly, which could be more annoying. (PS: I would not like to have any generated files too, with more reliable tools/packages/environments.)

**Review by sofroniewn — APPROVED** [review]
Looks good to me!

File: CONTRIBUTING.md (hunk 1 above)
Comment (jni) [reply CONTRIBUTING.md:66]: But this is the opposite of a single way of generating dev docs: with this system, they will be generated by any PR author (who touches the package structure) with their own system.

**Comment by AhmetCanSolak** [comment]
@jni :D i know, but wanted to put my reasoning so that you or others could show a better way if possible. Will be merging now but looking forward to hear more on this matter.

**Comment by jni** [comment]
I used [doctr](https://drdoctr.github.io) for [skan](https://jni.github.io/skan). It's great. Docs build on Travis.

**Pull request merged**